cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -897,7 +940,15 @@ TEST_F(TestS3FS, GetFileInfoGenerator) {
   // Non-root dir case is tested by generic tests
 }
 
-TEST_F(TestS3FS, GetFileInfoGeneratorStress) {
+class TestS3FSHTTP : public TestS3FS {
+ public:
+  void SetUp() override {
+    enable_tls_if_supported_ = false;
```

Comment:
Why was this necessary? This looks like a weird workaround, can there at least be an explanation in comments?

Reply:
I've temporarily disabled this workaround here: https://github.com/apache/arrow/pull/43601/commits/a69961bbd119a27a023961d38329bf01f5bc0f15

We'll see how CI goes

Reply:
It seems it can occur sporadically even in non-stress tests:
https://github.com/ursacomputing/crossbow/actions/runs/11388191999/job/31684559075#step:7:3791
```
[ RUN      ] TestS3FS.OpenOutputStreamMetadata
/arrow/cpp/src/arrow/filesystem/s3fs_test.cc:593: Failure
Failed
'output->Close()' failed with IOError: When completing multiple part upload for key 'mdfile1' in bucket 'bucket': AWS Error NO_SUCH_UPLOAD during CompleteMultipartUpload operation: The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.
/arrow/cpp/src/arrow/filesystem/s3fs.cc:1830  CleanupIfFailed(FinishPartUploadAfterFlush())
[  FAILED  ] TestS3FS.OpenOutputStreamMetadata (215 ms)
```

@kou Do you know if we were seeing this error before switching to HTTPS for Minio-based tests?

Reply:
I think this may have to do with forcing "localhost" for TLS certificate check purposes: in some cases, a port number might be reused?

What we should perhaps do is:
1) switch all tests to HTTP _except_ the SSE customer key tests
2) use "localhost" only for HTTPS tests, otherwise use `GetListenAddress()` without a host argument.

Reply:
@pitrou Fine, i've done the related change in the latest commit.  please help to review again.